### PR TITLE
fix Gtest memeory leak

### PR DESCRIPTION
**Pull request by jichangjichang**
for SWDEV-486715 and SWDEV-486721

### CODE REVIEW

File: clients/include/testing_auxiliary.hpp [hunk 1]
```diff
@@ -457,6 +462,12 @@ void testing_aux_get_sol_with_null_biasaddr(const Arguments& arg)
     CHECK_HIP_ERROR(hipFree(d_b));
     CHECK_HIP_ERROR(hipFree(d_c));
     CHECK_HIP_ERROR(hipFree(d_d));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatmulDescDestroy(matmul));
```
Comment (jichangjichang) [inline clients/include/testing_auxiliary.hpp:465]: The major fixes

File: clients/include/testing_auxiliary.hpp [hunk 2]
```diff
@@ -550,6 +560,12 @@ void testing_aux_get_sol_with_zero_alpha_null_a_b(const Arguments& arg)
 
     CHECK_HIP_ERROR(hipFree(d_c));
     CHECK_HIP_ERROR(hipFree(d_d));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatmulDescDestroy(matmul));
```
Comment (jichangjichang) [inline clients/include/testing_auxiliary.hpp:563]: The major fixes

File: clients/include/testing_auxiliary.hpp [hunk 3]
```diff
@@ -702,6 +717,11 @@ void testing_aux_matmul_alg_null_matmul(const Arguments& arg)
     CHECK_HIP_ERROR(hipFree(d_b));
     CHECK_HIP_ERROR(hipFree(d_c));
     CHECK_HIP_ERROR(hipFree(d_d));
+    CHECK_HIPBLASLT_ERROR(hipblasLtMatmulDescDestroy(matmul));
```
Comment (jichangjichang) [inline clients/include/testing_auxiliary.hpp:720]: The major fixes

File: clients/include/utility.hpp [hunk 4]
```diff
@@ -222,11 +222,21 @@ class hipblaslt_local_matmul_descr
...
+        hipblasLtMatmulDescSetAttribute(this->m_descr,
+                                        HIPBLASLT_MATMUL_DESC_COMPUTE_INPUT_TYPE_B_EXT,
+                                        &compute_input_typeB,
+                                        sizeof(void*));
     }
 
-    ~hipblaslt_local_matmul_descr() {}
+    ~hipblaslt_local_matmul_descr()
+    {
+        if(this->m_status == HIPBLAS_STATUS_SUCCESS)
```
Comment (jichangjichang) [inline clients/include/utility.hpp:237]: The major fixes

File: library/src/amd_detail/hipblaslt.cpp [hunk 5]
```diff
@@ -147,6 +142,11 @@ try
     int             deviceId;
     hipError_t      err;
     hipblasStatus_t retval = HIPBLAS_STATUS_SUCCESS;
+    // TODO: Synchronizer size pass into predicate SynchronizerSizeCheck
```
Comment (jichangjichang) [inline library/src/amd_detail/hipblaslt.cpp:145]: The major fixes

File: clients/gtest/hipblaslt_gtest_ext_op.cpp [hunk 6]
```diff
@@ -299,6 +299,8 @@ TEST_P(ExtOpLayerNormTest, layernormSuccess)
     err = hipFree(gpuMean);
     err = hipFree(gpuInvvar);
     err = hipFree(gpuInput);
+    err = hipFree(gpuGamma);
```
Comment (jichangjichang) [inline clients/gtest/hipblaslt_gtest_ext_op.cpp:302]: The major fixes

**Comment by jichangjichang** [comment]
Pass on gfx90a
[----------] Global test environment tear-down
[==========] 13061 tests from 13 test suites ran. (1805283 ms total)
[  PASSED  ] 13061 tests.
hipBLASLt version: 1000

**Comment by jichangjichang** [comment]
pass on gfx942
[----------] Global test environment tear-down
[==========] 48206 tests from 13 test suites ran. (1900706 ms total)
[  PASSED  ] 48206 tests.
hipBLASLt version: 1000

**Review by KKyang — APPROVED** [review]

**Review by hcman2 — APPROVED** [review]

**Pull request merged**